packages/calcite-components/src/utils/dom.ts

```diff
@@ -565,12 +576,25 @@ export function slotChangeHasAssignedElement(event: Event): boolean {
  * ```
  *
  * @param {Event} event The event.
- * @returns {boolean} Whether the slot has any assigned elements.
+ * @param {string} selector The CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
+ */
+export function slotChangeGetAssignedElements(event: Event, selector?: string): Element[] {
+  return getSlotAssignedElements(event.target as HTMLSlotElement, selector);
+}
+
+/**
+ * This helper returns the assigned elements on a `slot` element, filtered by an optional css selector.
+ *
+ * @param {HTMLSlotElement} slot The slot element.
+ * @param {string} selector CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
  */
-export function slotChangeGetAssignedElements(event: Event): Element[] {
-  return (event.currentTarget as HTMLSlotElement).assignedElements({
+export function getSlotAssignedElements<T>(slot: HTMLSlotElement, selector?: string): T[] {
```

Comment:
Let's make sure to add spec tests for the follow-up issue/PR we discussed earlier (updating similar functions with a new selector argument).

Reply:
Done.  Added a direct element selector, one with no selector, and a test for empty results and a css class selector.  That should cover the bases ⚾